docs/assessment.md

```diff
@@ -547,11 +547,13 @@ See the [`clientFiles` and `serverFiles`](clientServerFiles.md) page for details
 
 ## Student-attached personal notes
 
-Students can attach files to assessments as personal notes, either by uploading them or by pasting the file contents as text. This can be done on the assessment overview page, or on individual question pages. These files can be viewed by the student anytime they can view the assessment.
+By default, students can attach files to assessments as personal notes, either by uploading them or by pasting the file contents as text. This can be done on the assessment overview page, or on individual question pages. These files can be viewed by the student anytime they can view the assessment.
 
 The purpose of this is to allow students to take extra notes during exams, for later review. For example, if a student has a Matlab script that they used to solve a question, they could attach it to that question so they can review it later.
 
-This file attachment functionality does not provide a way for students to attach files before an exam starts, so it can't be used for student-provided "formula sheets" on exams.
+Although we do not recommend this for regular exams, personal notes can be disabled by setting `"allowPersonalNotes": false` as a top-level option in the `infoAssessment.json` file. Note that this setting only prevents the uploading or modification of notes, but does not delete or prevent access to existing notes that were uploaded before the option is set.
```

Comment:
> Note that this setting only prevents the uploading or modification of notes, but does not delete or prevent access to existing notes that were uploaded before the option is set.

This is not how I'd expect this to behave. Can you say more about why you picked this behavior?

Reply:
As mentioned above, the main reason is that this is what `allowNewUploads` current does as implemented by the PersonalNotesPanel. In practice, I don't think that the handling of existing notes really matters unless an instructor first (accidentally?) allows notes and then disables them after the fact.

If you think this is worth changing to not display notes at all, I could either add a new option or change the implementation of the current one. Before this change, the only place where `allowNewUploads` is ever set to false is in `instanceQuestion.html.ts`, and to be honest I don't quite see how it would ever be displayed there since that requires `resLocals.file_list.length > 0` (so an existing personal note that the instructor somehow uploaded?).

Reply:
The `instanceQuestion.html.ts` you refer to is manual grading. This is used to allow graders to see personal notes added by students, but not upload new ones.

Reply:
I don't think it makes much sense to show the personal notes section if it won't contain notes or buttons to upload new ones. Rather than using `allowNewUploads`, I'd just conditionally render the entire `PersonalNotesPanel` based on this new setting.

Note that you'll also need to update the server-side upload handling to reject uploads if they're not allowed for the current assessment. Here's one location (there are others): https://github.com/PrairieLearn/PrairieLearn/blob/9af9e7c48ebb23924f9438a124abf8aa7a979e7a/apps/prairielearn/src/pages/studentInstanceQuestion/studentInstanceQuestion.ts#L58-L58

Reply:
Ah I see. I think someone mentioned during the OHs that personal notes are not visible to instructors, but I guess they are through this page. Alright, so it sounds like changing `allowNewUploads` is not the way to go then, thanks!

Reply:
Personal notes are visible to instructors, even outside the context of manual grading. The only restriction is that instructions cannot (by default) upload new files (though they can if they use an effective user).

Reply:
Updated to make the entire box render conditionally and to prevent any server-side actions and existing file downloads if the flag is disabled.